diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 3f25517..d41c833 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -67,7 +67,7 @@ configure_file(
 target_include_directories(unifex
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include/>
-    "${PROJECT_BINARY_DIR}/include")
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>)
 
 target_compile_features(unifex PUBLIC cxx_std_17)
 
